src/webgpu/shader/execution/padding.spec.ts

```diff
@@ -421,3 +421,85 @@ g.test('vec3')
     `;
     runShaderTest(t, wgsl, new Uint32Array([0x12345678, 0xabcdef01, 0x98765432, 0xdeadbeef]));
   });
+
+g.test('matCx3')
+  .desc(
+    `Test padding bytes are preserved when assigning to a variable of type matCx3.
+    `
+  )
+  .params(u =>
+    u
+      .combine('columns', [2, 3, 4] as const)
+      .combine('use_struct', [true, false] as const)
+      .beginSubcases()
+  )
+  .fn(t => {
+    const cols = t.params.columns;
+    const wgsl = `
+      alias Mat = mat${cols}x3<f32>;
+      ${t.params.use_struct ? `struct S { m : Mat } alias Type = S;` : `alias Type = Mat;`}
+      @group(0) @binding(0) var<storage, read_write> buffer : Type;
+
+      @compute @workgroup_size(1)
+      fn main() {
+        var m : Mat;
+        for (var c = 0u; c < ${cols}; c++) {
+          m[c] = vec3(f32(c*3 + 1), f32(c*3 + 2), f32(c*3 + 3));
+        }
+        buffer = Type(m);
+      }
+    `;
+    const f_values = new Float32Array(16);
```

Comment:
übernit: You could change `16` to `cols * 4`, then drop the slice. Took me a while to figure out what the slice was for!

Reply:
I have no idea why I wrote this with a `slice`.

Done.